From f95c4255fa419fa706becff476558aeee07fdfce Mon Sep 17 00:00:00 2001
From: "Yang, Dong" <dong.yang@intel.com>
Date: Fri, 9 Jun 2023 21:30:15 +0530
Subject: [PATCH] INTERNAL: Add more usage flag for DRM_FORMAT_ABGR2101010

The DRM_FORMAT_ABGR2101010 format will be used by Android Media
framework for software codec of AV1Hdr, and the usage flag request
as:
 BO_USE_SW_READ_RARELY
 BO_USE_SW_READ_OFTEN
 BO_USE_SW_WRITE_RARELY
 BO_USE_SW_WRITE_OFTEN

Tracked-On: OAM-95578
Signed-off-by: Yang, Dong <dong.yang@intel.com>
---
 i915.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/i915.c b/i915.c
index a6a2cdc..dabdced 100644
--- a/i915.c
+++ b/i915.c
@@ -256,6 +256,9 @@ static int i915_add_combinations(struct driver *drv)
 
 	/* Android CTS tests require this. */
 	drv_add_combination(drv, DRM_FORMAT_BGR888, &metadata_linear, BO_USE_SW_MASK);
+#ifdef USE_GRALLOC1
+	drv_modify_combination(drv, DRM_FORMAT_ABGR2101010, &metadata_linear, BO_USE_SW_MASK);
+#endif
 
 	/*
 	 * R8 format is used for Android's HAL_PIXEL_FORMAT_BLOB and is used for JPEG snapshots
-- 
2.40.1

